### ISSUE TITLE
Add support for omitnil

### ISSUE THREAD

**Issue by ivanjaros** [go-json-experiment/json#53]
I have been fighting Go's json for years when it comes to handling of null values.
Go's implementation only has `omitempty` and it treats `{}`, `[]` and `null` as one and the same. But `{}` and `[]` are information of intent whereas `null` is not. I have been handling this issue in various ways and presently I am using a fork of native json library, after years of using Goccy and clean-up code that stripped null values from raw json. This was still more performant than using native json. Sadly the entire standard library lacks in performance, after a decade of neglect. But goccy has been lagging with issues and I no longer trust it, hence the fork.

In short, please add **configuration option** to not render null values. Also add `omitnil` or `omitnull` tag, so we can target maps and slices individually when we want to distinguish nil and empty values, since maps and slices are pointers in Go.

**Comment by zzhaolei** [comment]
I think this is related to https://github.com/golang/go/issues/45669

**Comment by dsnet** [comment]
I suspect that your use-case is already satisfied by the `omitzero` option that that originated in this module and has since been back-ported to the "encoding/json" package for the upcoming release of Go.

**Comment by ivanjaros** [comment]
Technically it solves my issue but only on struct tag basis. I want a global configuration targeting only nils specifically, not all zero values.

**Comment by dsnet** [comment]
#61 adds a caller-specified option to omit all zero struct fields.

**Comment by ivanjaros** [comment]
global omitzero will get rid of false or 0, i don't know how else to communicate i want to get rid of null values in json, not default values in Go.

i'll stick with my forked native json because this brings no value to me. cheers.
